### PR TITLE
Part 2

### PR DESCRIPTION
**Pull request by ruancomelli**
## Summary by Sourcery

Introduce a new 'Class Distribution Tool' page for managing class distributions, refactor the 'create_classes' function, update 'Department' field options across various JSON files, and clean up commented-out code.

New Features:
- Introduce a new 'Class Distribution Tool' page for managing class distributions, including form fields for group title, year, stage, faculty, department, time, and semester, along with buttons to fetch students and create classes.

Enhancements:
- Refactor the 'create_classes' function to accept individual parameters instead of a document object.
- Update various JSON files to change the 'Department' field options from 'Faculty Department' to 'Department'.
- Make the 'Faculty' field in 'Presented Module' read-only.

Chores:
- Remove commented-out code from 'group_class.js' and 'student_portal.js'.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is an experiment review for experiment `improve_review_guide`
Run ID: `improve_review_guide/run_2024-08-12T00-19-10_v1-21-0-80-gb1a031cd3-dirty`
The benchmark review for this pull request can be found at https://github.com/sourcery-ai-experiments/kalima/pull/17.
This pull request was cloned from `https://github.com/sourcery-ai-experiments/kalima/pull/3`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_

<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false

    enable_claude_comments: true
    enable_functionality_review: false

    enable_pull_request_summary: true
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      docstrings_model: benchmark
      functionality_model: benchmark
      overall_comments_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: llm
      review_guide_model: llm

# The pull request dataset to run the experiment on
pull_request_dataset:
  # open-source PRs
- https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
- https://github.com/sourcery-ai-experiments/iptux/pull/5
- https://github.com/sourcery-ai-experiments/iptux/pull/6
- https://github.com/sourcery-ai-experiments/kalima/pull/3
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
- https://github.com/sourcery-ai-experiments/Trenova/pull/11
  # Sourcery PRs
- https://github.com/sourcery-ai/core/pull/4055
- https://github.com/sourcery-ai/core/pull/4263
- https://github.com/sourcery-ai/core/pull/4352
- https://github.com/sourcery-ai/core/pull/4364
- https://github.com/sourcery-ai/core/pull/4366
- https://github.com/sourcery-ai/core/pull/4390
- https://github.com/sourcery-ai/core/pull/4402
- https://github.com/sourcery-ai/core/pull/4445
- https://github.com/sourcery-ai/core/pull/4462
- https://github.com/sourcery-ai/core/pull/4463
- https://github.com/sourcery-ai/core/pull/4471

review_comment_labels: []

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews:
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4463
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/236
- dataset_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/gpt-crawler/pull/10
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/5
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/32
- dataset_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/6
  review_pull_request: https://github.com/sourcery-ai-experiments/iptux/pull/33
- dataset_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/3
  review_pull_request: https://github.com/sourcery-ai-experiments/kalima/pull/17
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4055
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/258
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4263
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/259
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4352
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/260
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4364
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/261
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4366
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/262
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4390
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/263
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4402
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/264
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4445
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/265
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4462
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/266
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/23
- dataset_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/24
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/19
- dataset_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
  review_pull_request: https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/20
- dataset_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/11
  review_pull_request: https://github.com/sourcery-ai-experiments/Trenova/pull/19
- dataset_pull_request: https://github.com/sourcery-ai/core/pull/4471
  review_pull_request: https://github.com/sourcery-ai-experiments/core/pull/277

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request introduces a new 'Class Distribution Tool' page and makes several updates to existing files, primarily focusing on standardizing department references and refactoring the class creation process. The changes include:

1. A new 'Class Distribution Tool' page (class_distribution_t.js) for managing class distributions.
2. Refactoring of the 'create_classes' function in group_class.py to accept individual parameters instead of a document object.
3. Updates to multiple JSON files to replace 'Faculty Department' with 'Department' for consistency.
4. Modification of the 'Graduation Year' field in the 'Student' doctype to be a selectable list of years.
5. Various minor updates and adjustments to existing files.

### File-Level Changes

| Files | Changes |
| ----- | ------- |
| `kalima/kalima/doctype/student_scores/student_scores.json`<br/>`kalima/kalima/doctype/presented_module/presented_module.json`<br/>`kalima/kalima/doctype/class/class.json`<br/>`kalima/kalima/doctype/applicant_student/applicant_student.js`<br/>`kalima/kalima/page/student_result_sheet_1/student_result_sheet_1.js`<br/>`kalima/kalima/doctype/activity_departments/activity_departments.json`<br/>`kalima/kalima/doctype/applicant_student/applicant_student.json`<br/>`kalima/kalima/doctype/complaints/complaints.json`<br/>`kalima/kalima/doctype/constant/constant.json`<br/>`kalima/kalima/doctype/continuous_exam_result/continuous_exam_result.json`<br/>`kalima/kalima/doctype/department_module/department_module.json`<br/>`kalima/kalima/doctype/exam_departments/exam_departments.json`<br/>`kalima/kalima/doctype/exam_result_entry/exam_result_entry.json`<br/>`kalima/kalima/doctype/exam_students/exam_students.json`<br/>`kalima/kalima/doctype/group_class/group_class.json`<br/>`kalima/kalima/doctype/library/library.json`<br/>`kalima/kalima/doctype/outgoing/outgoing.json`<br/>`kalima/kalima/doctype/preferred_departments/preferred_departments.json`<br/>`kalima/kalima/doctype/presented_department/presented_department.json`<br/>`kalima/kalima/doctype/student_attendance_entry/student_attendance_entry.json`<br/>`kalima/kalima/doctype/student_result_log/student_result_log.json`<br/>`kalima/kalima/doctype/students_fees/students_fees.json`<br/>`kalima/kalima/doctype/transfer/transfer.json` | Updated 'Department' references: Changed 'Faculty Department' to 'Department' in various JSON files for consistency across the system. |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: kalima/kalima/doctype/group_class/group_class.py [hunk 1]
```diff
@@ -60,17 +62,21 @@ def fetch_students(selected_modules, department):
...
-
-
 @frappe.whitelist()
-def create_classes(group_class_doc,group_class_modules,students):
+def create_classes(group_title,year,stage,semester,department,group_class_modules,students):
     group_class_modules = json.loads(str(group_class_modules))
     for module in group_class_modules:
         # Convert the group_class_doc to a dictionary
-        group_class_doc = json.loads(str(group_class_doc))
+        # group_class_doc = json.loads(str(group_class_doc))
```
Comment (SourceryAI) [inline kalima/kalima/doctype/group_class/group_class.py:70]: **suggestion:** Remove commented-out code if it's not needed.

Commented-out code clutters the codebase and can be confusing. If it's not needed, it should be removed.

File: kalima/kalima/page/class_distribution_t/class_distribution_t.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+            {
+                fieldtype: 'Link',
+                fieldname: 'department',
+                label: 'Department',
+                // options: 'Faculty Department',
+                options: 'Department',
+                onchange: function () {
+                    let stage = form.get_value('stage');
+                    let department = form.get_value('department');
+                    let faculty = form.get_value('faculty');
```
Comment (SourceryAI) [inline kalima/kalima/page/class_distribution_t/class_distribution_t.js:48]: **suggestion (performance):** Store form values in variables to optimize repeated use of `form.get_value`.

Repeatedly calling `form.get_value` can be optimized by storing the values in variables. This makes the code cleaner and potentially faster.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes - here's some feedback:

**Overall Comments**:
- Remove commented-out code to keep the codebase clean.
- Consider modularizing the JavaScript code in the new `Class Distribution Tool` page for better maintainability.
- Use templates or a more dynamic approach to generate HTML content in the `fetch_students` function.

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 2 issues found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/e05138cc-c1a1-4976-8562-8e649059407f?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**